packages/synapse-react-client/src/components/FullTextSearch/FullTextSearch.test.tsx

```diff
@@ -0,0 +1,188 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import React from 'react'
+import { FullTextSearch } from './FullTextSearch'
+import {
+  QueryVisualizationContextProvider,
+  QueryVisualizationContextType,
+} from '../QueryVisualizationWrapper'
+import {
+  QueryContextProvider,
+  QueryContextType,
+} from '../QueryContext/QueryContext'
+import {
+  ColumnSingleValueFilterOperator,
+  QueryBundleRequest,
+} from '@sage-bionetworks/synapse-types'
+import {
+  mockQueryBundleRequest,
+  mockQueryResultBundle,
+} from '../../mocks/mockFileViewQuery'
+import { server } from '../../mocks/msw/server'
+import { registerTableQueryResult } from '../../mocks/msw/handlers/tableQueryService'
+import { updateQueryUsingSearchTerm } from './FullTextSearchUtils'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+
+const initialQueryRequest = mockQueryBundleRequest
+
+const renderComponent = (
+  queryContext: Partial<QueryContextType>,
+  queryVisualizationContext: Partial<QueryVisualizationContextType>,
+) => {
+  return render(
+    <QueryContextProvider queryContext={queryContext as QueryContextType}>
+      <QueryVisualizationContextProvider
+        queryVisualizationContext={
+          queryVisualizationContext as QueryVisualizationContextType
+        }
+      >
+        <FullTextSearch />
+      </QueryVisualizationContextProvider>
+    </QueryContextProvider>,
+    {
+      wrapper: createWrapper(),
+    },
+  )
+}
+
+const mockExecuteQueryRequest = jest.fn()
+const mockGetLastQueryRequest = jest.fn()
+
+const queryContext: Partial<QueryContextType> = {
+  executeQueryRequest: mockExecuteQueryRequest,
+  queryMetadataQueryOptions: {
+    queryKey: ['queryMetadataQueryOptions'],
+    queryFn: jest.fn().mockResolvedValue(mockQueryResultBundle),
+  },
```

Comment:
Have to mock `queryMetadataQueryOptions` now because `FullTextSearch` uses it (in place of the Jotai atom)